internal/yggdrasil/yggdrasil.go

```diff
@@ -234,6 +239,29 @@ func (h *Handler) PostDataMessageForDevice(ctx context.Context, params yggdrasil
 	return operations.NewPostDataMessageForDeviceOK()
 }
 
+func (h *Handler) updateDeviceStatus(ctx context.Context, device *v1alpha1.EdgeDevice, updateFunc func(d *v1alpha1.EdgeDevice)) middleware.Responder {
+	patch := client.MergeFrom(device.DeepCopy())
+	updateFunc(device)
+	err := h.deviceRepository.PatchStatus(ctx, device, &patch)
+	if err == nil {
+		return nil
+	}
+
+	// retry patching the edge device status
+	for i := 0; i < 3; i++ {
```

Comment:
What do you think about pausing between retries? 
Exponential back-off might be helpful if there is a lot of changes happening to the device at the same time.